public/app/features/apiserver/client.ts

```diff
@@ -40,18 +40,26 @@ export class ScopedResourceClient<T = object, S = object, K = string> implements
     return getBackendSrv().get<Resource<T, S, K>>(`${this.url}/${name}`);
   }
 
-  public watch(opts?: WatchOptions): Observable<ResourceEvent<T, S, K>> {
+  public watch(
+    params?: WatchOptions,
+    config?: Pick<BackendSrvRequest, 'data' | 'method'>
+  ): Observable<ResourceEvent<T, S, K>> {
     const decoder = new TextDecoder();
-    const params = {
-      ...opts,
+    const { name, ...rest } = params ?? {}; // name needs to be added to fieldSelector
+    const requestParams = {
+      ...rest,
       watch: true,
-      labelSelector: this.parseListOptionsSelector(opts?.labelSelector),
-      fieldSelector: this.parseListOptionsSelector(opts?.fieldSelector),
+      labelSelector: this.parseListOptionsSelector(params?.labelSelector),
+      fieldSelector: this.parseListOptionsSelector(params?.fieldSelector),
     };
+    if (name) {
+      requestParams.fieldSelector = `metadata.name=${name}`;
```

Comment:
previous code thought you could watch the specific name -- -but it must be sent as a fieldSelector